sakura_core/charset/codechecker.h

```diff
@@ -388,12 +388,18 @@ char32_t ConvertToUtf32(std::wstring_view text) {
 	return 0;
 }
 
+/*!
+ * 文字列がIVSの異体字セレクタで始まっているか判定する
+ * nLenを省略した時はヌル文字の手前までが有効な文字列であると解釈する
+ */
+inline bool IsVariationSelector(const wchar_t* pStr, size_t nLen = SIZE_MAX) {
+	return (2 <= nLen) && (pStr[0] == 0xDB40) && (0xDD00 <= pStr[1]) && (pStr[1] <= 0xDDEF);
```

Comment:
```suggestion
	return IsVariationSelector(std::wstring_view(pStr, nLen));
```
こうしておくとpStr,nLenに不正値が入っていた場合（＝プログラムのバグ）を検出できます。